facet-reflect/Cargo.toml

```diff
@@ -12,16 +12,21 @@ categories = ["development-tools", "rust-patterns"]
 
 [features]
 default = ["std"]
-std = ["alloc"]
+std = ["alloc", "indexmap"]
 alloc = []
 
 [dependencies]
+facet-ansi = { path = "../facet-ansi" }
 facet-core = { path = "../facet-core", version = "0.5.3" }
+indexmap = { version = "2.9.0", optional = true }
```

Comment:
This prevents `Wip` being used in non-std contexts

Is the dual nature of ordered and indexable by custom type required? Or is it a nicety, like order preserving?

Options
- Switch to `BTreeMap` in `alloc` contexts
- Pull in a custom hash implementation to still use `indexmap` (but then you can't deactivate that dep in `std` contexts)
- Implement a custom "index map" using `Vec` (e.g. clap does this with [`flat_map`](https://github.com/clap-rs/clap/blob/master/clap_builder/src/util/flat_map.rs) but the motivation was for binary size)

Reply:
> Or is it a nicety, like order preserving?

It's a correctness thing, because we need to be able to drop fields in the right order, but I don't think we actually need an index map for that. 

I'm also not exactly sure what the drop order should be for partially initialized structs, to be honest??